rfcs/rfc-0000-interactions.md

```diff
@@ -0,0 +1,599 @@
+- Feature Name: Interactions
+- Start Date: 06/04/2023
+- RFC PR: https://github.com/revoltchat/rfcs/issues/0000
+- Tracking Issue: https://github.com/revoltchat/revolt/issues/0000
+- Status: draft
+
+# Summary
+
+Interactions are a way of interacting with bots in a higher level with better intergration
+with revolt and the UI. This will include, slash commands, message blocks, context menu commands,
+customizable config page and a dashboard.
+
+# Motivation
+
+A large part of developing a bot is handling a lot underlying parts like command handling and
+a way to let users configure the bot for there server, as both of these can be done in a variety
+of ways it can cause a lot of fragmentation in the ways its done, different bots could parse
+commands in different ways making it hard for the user to remember how each bot handles it,
+increasing cognitive load on the user, along with this is the fact that there is no UI to support
+it as its all handled by the bot. In a similar case configuring the bot can be either done via
+commands or via a website which requires the user to leave revolt.
+
+# Guide-level explanation
+
+## Slash command
+
+A defined command users can invoke to trigger actions from a bot, the parameters are known and
+is intergrated into the UI in a way that the user can understand what the command takes before
+they run it and while writing it. Invoking a slash commands requires typing "/" into the message
+box. Slash commands are never sent as a message and are instead directly sent as a seperate request,
+other people will not see the command being sent unless the response is a message.
+
+Slash commands are able to have parameters with a variety of types and subcommands to group commands
+together.
+
+parameters are able to have autocomplete to allow entering certain types of infomation easier.
+
+## Context Menu Commands
+
+Context menu commands are actions you can invoke on any message or user from the context menu
+(right click or click and hold on mobile), these allow users to run actions on a message or user.
+
+member context menu commands are not attached to any specific channel and are invoked at a server level,
```

Comment:
would it be possible to allow channels to be included in this type of interaction regardless? i think it could be useful, especially for bots that deal with permissions management.

Reply:
I cant see how that would work, you are not always in one channel or even the same channel, what if i tried to go through this when in settings, or i clicked a member which is in a voice channel? should it go off the channel im viewing or the voice channel they are in?

Reply:
maybe it could just be an optional parameter?

or maybe there could even be some kind of "context" parameter, which lists exactly what menu *or* channel you're calling the command from. (although admittedly i cant see how this would be any more useful lmao. just spitballing)